### PR TITLE
GRAPHICS: Make HQx ASM scalers relocations free

### PR DESCRIPTION
**Pull request by lephilousophe**
This is considered as a good practice to not have relocations in text segment.
The resulting binary is now completely PIC and can easily be relocated at any address (ASLR).

After a quick benchmark done on real game, the performance seems to be equivalent if not faster to the older assembly code.
The C++ implementation is still around 3 times slower even in x86_64 mode (so with SSE2 extensions enabled).

### CODE REVIEW

**Comment by ccawley2011** [comment]
Would it be possible to move the static variables `hqx_params` and `RGBtoYUV` into the `HQScaler` class? The HQx scalers are currently the only ones that don't support having multiple instances following PR #3557.

**Comment by lephilousophe** [comment]
@ccawley2011 Do you mean like this? :wink:

**Review by ccawley2011 — APPROVED** [review]

**Comment by aquadran** [comment]
it would be nice to add 64bits calling convention in future

**Comment by lephilousophe** [comment]
> it would be nice to add 64bits calling convention in future
The whole asm code is using 32 bits registers.
I would say that the code should be rewritten with x86_64 registers in mind and with SSE2 instructions in another file.
This may improve performance.

**Comment by aquadran** [comment]
I'm not sure if it's worthy rewrite it, I mean, not sure improvements would be. I'm just saying use it what we have at least.

File: graphics/scaler/hq.cpp [hunk 1]
```diff
@@ -4987,72 +4947,90 @@ static void HQ3x_implementation(const uint8 *srcPtr, uint32 srcPitch, uint8 *dst
...
 		                               11, 5, 0, 0);
-		InitLUT(format16);
+		initLUT(format16);
 	}
 }
 
 HQScaler::~HQScaler() {
-	free(RGBtoYUV);
-	RGBtoYUV = 0;
+	delete _RGBtoYUV;
```
Comment (mduggan) [inline graphics/scaler/hq.cpp:4969]: I think this should be delete []?

It almost never makes a difference for basic types but nice to be correct.

**Comment by bluegr** [comment]
Nice work making the code PIC (Position-independent)!

File: graphics/scaler/hq.cpp (hunk 1 above)
Comment (lephilousophe) [reply graphics/scaler/hq.cpp]: Good catch!
Thank you very much.

**Comment by lephilousophe** [comment]
Merging now

**Pull request merged**